go-controller/pkg/ovn/egressgw.go

```diff
@@ -481,7 +485,7 @@ func (oc *Controller) deleteGWRoutesForPod(name ktypes.NamespacedName, podIPNets
 		if gwToGr, ok := routeInfo.podExternalRoutes[pod]; ok {
 			if len(gwToGr) == 0 {
 				delete(routeInfo.podExternalRoutes, pod)
-				return
+				continue
```

Comment:
this looks like a separate bug fix right? Can we move this into its own commit. May also need bz for this?

Reply:
done